src/mrpro/data/AcqInfo.py

```diff
@@ -6,23 +6,24 @@
 import ismrmrd
 import numpy as np
 import torch
-from typing_extensions import Self, TypeVar
+from einops import rearrange
+from typing_extensions import Self
 
 from mrpro.data.MoveDataMixin import MoveDataMixin
+from mrpro.data.Rotation import Rotation
 from mrpro.data.SpatialDimension import SpatialDimension
+from mrpro.utils.unit_conversion import mm_to_m
 
-# Conversion functions for units
-T = TypeVar('T', float, torch.Tensor)
 
+def rearrange_acq_info_fields(field: object, pattern: str, **axes_lengths: dict[str, int]) -> object:
+    """Change the shape of the fields in AcqInfo."""
+    if isinstance(field, Rotation):
```

Comment:
why only Rotation and torch.Tensor? Will this also cover SpatialDimension and AcqIdx?

Reply:
Both have a apply_ on its own and we will recurse into them. Only rotation and tensor are things we need to reshape.

Reply:
and once you recurse into them than the fields of SpatialDimension and AcqIdx will be torch.Tensors?

Reply:
Yes

Reply:
That seems to work 😜